File: Algorithms/Challenges/Strings/LexicographicRotation/Python/LexicographicRotationTest.py
import unittest
import LexicographicRotation as lr

class LexicographicRotationTest(unittest.TestCase):

    def test_basic(self):
        string = 'rotate'
        self.assertEqual(lr.getMaxRotation(string), 'terota')
        self.assertEqual(lr.getMinRotation(string), 'aterot')

    def test_medium(self):
        string = 'abcdefghijklmnopqrstuvwxyz'
        self.assertEqual(lr.getMaxRotation(string), 'zabcdefghijklmnopqrstuvwxy')
        self.assertEqual(lr.getMinRotation(string), 'abcdefghijklmnopqrstuvwxyz')

    def test_duplicates(self):
        string = 'xaxbxexf'
        self.assertEqual(lr.getMaxRotation(string), 'xfxaxbxe')
        self.assertEqual(lr.getMinRotation(string), 'axbxexfx')

if __name__ == '__main__':
    unittest.main()

File: Algorithms/Challenges/Strings/LexicographicRotation/Java/LexicographicRotationTest.java
import org.junit.Test;
import static org.junit.Assert.*;

public class LexicographicRotationTest {
  private static final double DELTA = 1e-15;

  @Test
  public void basicTest() {
    LexicographicRotation lr = new LexicographicRotation();
    String string = "rotate";
    assertEquals(lr.getMaxRotation(string), "terota");
    assertEquals(lr.getMinRotation(string), "aterot");
  }

  @Test
  public void mediumTest() {
    LexicographicRotation lr = new LexicographicRotation();
    String string = "abcdefghijklmnopqrstuvwxyz";
    assertEquals(lr.getMaxRotation(string), "zabcdefghijklmnopqrstuvwxy");
    assertEquals(lr.getMinRotation(string), "abcdefghijklmnopqrstuvwxyz");
  }

  @Test
  public void duplicatesTest() {
    LexicographicRotation lr = new LexicographicRotation();
    String string = "xaxbxexf";
    assertEquals(lr.getMaxRotation(string), "xfxaxbxe");
    assertEquals(lr.getMinRotation(string), "axbxexfx");
  }
}
